### ISSUE TITLE
Uncaught TypeError: Cannot read property 'data' of undefined

### ISSUE THREAD

**Issue by nlarche** [TylorS/snabbdom-selector#29]
Hello,

I've got this error from v3.0.0 when the virtual dom tree contains undefined vnodes : 

```
Uncaught TypeError: Cannot read property 'data' of undefined
      at customMatches (node_modules/snabbdom-selector/src/query.ts:34:11)
```

this happen when I conditionnaly render a vnode or not

`if (condition) ? h('div'): undefined;`

I can return an empty string instead of undefined to workaround.

thanks

**Comment by nlarche** [comment]
Hello @atomrc, I need this to update to the last version of @cycle/dom in my application, when do you think you're going to update the dependency ?

**Comment by atomrc** [comment]
@nlarche Since the version declared in the package.json is `^2.0.0`, I believe if you reinstall `snabbdom-selector` it should pull the latest version of `tree-selector` too. 

If it is not enough, I also have a branch somewhere with some other improvements for this scenario. Will try to merge soon

**Comment by atomrc** [comment]
@nlarche I published version `4.1.0` which includes improvements to allow `undefined` values.
Please let me know if this fixed you issue.

**Comment by nlarche** [comment]
hi @atomrc, thanks issue is fixed ! 😄
